src/io/flutter/inspector/HeapDisplay.java

```diff
@@ -82,153 +82,147 @@ public void dispose() {
     @Override
     public void update(List<IsolateObject> isolates) {
       heapState.update(isolates);
+
       for (HeapDisplay graph : graphs) {
-        // TODO(pq): stop storing points in the graph and just use the samples directly.
-        heapState.getSamples().forEach(sample -> graph.addMeasure(scale(heapState, sample)));
+        graph.updateFrom(heapState);
       }
+
       panels.forEach(panel -> SwingUtilities.invokeLater(panel::repaint));
     }
   }
 
-  /**
-   * A fixed-length list of captured samples.
-   */
-  public static class HeapSamples {
-    private final LinkedList<HeapSample> samples = new LinkedList<>();
-    private final int capacity;
+  private static final int TEN_MB = 1024 * 1024 * 10;
 
-    HeapSamples(int size) {
-      this.capacity = size;
-    }
+  private static final int PREFFERED_WIDTH = 60;
+  private static final int PREFFERED_HEIGHT = 16;
 
-    void add(HeapSample sample) {
-      if (samples.size() == capacity) {
-        samples.removeFirst();
-      }
-      samples.add(sample);
-    }
+  private static final Stroke GRAPH_STROKE = new BasicStroke(2f);
 
-    int size() {
-      return samples.size();
-    }
+  private static final DecimalFormat df = new DecimalFormat();
 
-    HeapSample get(int index) {
-      return samples.get(index);
-    }
+  static {
+    df.setMaximumFractionDigits(1);
   }
 
-  public static class HeapState implements HeapListener {
-    // Running count of the max heap (in bytes).
-    private int heapMaxInBytes;
-
-    private final HeapSamples samples;
+  private @Nullable HeapState heapState;
 
-    HeapState(int sampleCapacity) {
-      samples = new HeapSamples(sampleCapacity);
-    }
+  public HeapDisplay() {
+    setVisible(true);
+  }
 
-    public Iterable<HeapSample> getSamples() {
-      return samples.samples;
-    }
+  private void updateFrom(HeapState state) {
+    this.heapState = state;
 
-    public int getMaxHeapInBytes() {
-      return heapMaxInBytes;
+    if (!heapState.getSamples().isEmpty()) {
+      final HeapSample sample = heapState.getSamples().get(0);
+      setToolTipText(printMb(sample.getBytes()) + " of " + printMb(heapState.getMaxHeapInBytes()));
```

Comment:
Neat!

I was imagining we'd display this next to the sample graph but I guess that depends on how much horizontal space we have.  Great idea to put it in the tooltip. 👍

Reply:
Sounds great! We can remove this this tooltip then.